src/librustc/ty/query/job.rs

```diff
@@ -349,6 +350,26 @@ fn connected_to_root<'tcx>(
     }).is_some()
 }
 
+// Deterministically pick an query from a list
+#[cfg(parallel_queries)]
+fn pick_query<'a, 'tcx, T, F: Fn(&T) -> (Span, Lrc<QueryJob<'tcx>>)>(
+    tcx: TyCtxt<'_, 'tcx, '_>,
+    queries: &'a [T],
+    f: F
+) -> &'a T {
+    // Deterministically pick an entry point
+    // FIXME: Sort this instead
+    let mut hcx = tcx.create_stable_hashing_context();
+    queries.iter().min_by_key(|v| {
+        let (span, query) = f(v);
+        let mut stable_hasher = StableHasher::<u64>::new();
+        query.info.query.hash_stable(&mut hcx, &mut stable_hasher);
+        let no_span = span == DUMMY_SP;
+        // Prefer entry points which have valid spans for nicer error messages
+        (no_span, stable_hasher.finish())
```

Comment:
This looks like things with no span would be preferred to things with span? Can you make `no_span` an integer so that one can easier see what's going on?

Reply:
Is `let no_span = (span == DUMMY_SP) as u8` any better?

Reply:
I'd prefer something like:
```rust
// Sort span-less queries to the back
let span_cmp = if span == DUMMY_SPAN { 1 } else { 0 };
```